plugins/feedback-backend/src/service/router.ts

```diff
@@ -108,31 +109,38 @@ export async function createRouter(
         const type = annotations['feedback/type'];
         const replyTo = annotations['feedback/email-to'];
         const reporterEmail = (
-          (await catalogClient.getEntityByRef(reqData.createdBy!, { token }))
-            ?.spec?.profile as { email: string }
-        ).email;
+          (await catalogClient.getEntityByRef(reqData.createdBy!, {
+            token,
+          })) as UserEntityV1alpha1
+        ).spec.profile?.email;
         const appTitle = config.getString('app.title');
 
         if (
           type.toUpperCase() === 'JIRA' &&
           !reqData.tag?.match(/(Excellent|Good)/g)
         ) {
           let host = annotations['feedback/host'];
+          let serviceConfig: Config;
           // if host is undefined then
           // use the first host from config
-          const serviceConfig =
-            config
-              .getConfigArray('feedback.integrations.jira')
-              .find(hostConfig => host === hostConfig.getString('host')) ??
-            config.getConfigArray('feedback.integrations.jira')[0];
+          try {
+            serviceConfig =
+              config
+                .getConfigArray('feedback.integrations.jira')
+                .find(hostConfig => host === hostConfig.getString('host')) ??
+              config.getConfigArray('feedback.integrations.jira')[0];
+          } catch {
+            return logger.error('Jira integeration not found');
+          }
           host = serviceConfig.getString('host');
           const authToken = serviceConfig.getString('token');
           const hostType = serviceConfig.getOptionalString('hostType');
 
           const projectKey = entityRef.metadata.annotations['jira/project-key'];
           const jiraService = new JiraApiService(host, authToken, hostType);
-          const jiraUsername =
-            await jiraService.getJiraUsernameByEmail(reporterEmail);
+          const jiraUsername = reporterEmail
+            ? await jiraService.getJiraUsernameByEmail(reporterEmail)
```

Comment:
should this await be under try catch block?

Reply:
it can be under **try-catch**, but its not required as the function won't be called if reporterEmail is `undefined`